src/core/Akka/Util/AppVersion.cs

```diff
@@ -32,20 +34,50 @@ namespace Akka.Util
     public class AppVersion : IComparable<AppVersion>, IEquatable<AppVersion>
     {
         public static readonly AppVersion Zero = new AppVersion("0.0.0");
+
+        // made internal for testing purposes
+        internal const string AssemblyVersionMarker = "assembly-version";
         private const int Undefined = 0;
 
-        private int[] numbers = Array.Empty<int>();
-        private string rest = "";
+        private int[] _numbers = Array.Empty<int>();
+        private string _rest = "";
 
         [JsonConstructor]
         internal AppVersion(string version)
         {
-            this.Version = version;
+            Version = version;
         }
 
         public static AppVersion Create(string version)
         {
-            var v = new AppVersion(version);
+            // check to see if we're going to use the assembly-version
+            if (version.Equals(AssemblyVersionMarker, StringComparison.InvariantCultureIgnoreCase))
```

Comment:
Check to see if we're using `assembly-version` - if so, get the assembly data. If not, parse the version string as usual.